app/controllers/addresses_controller.rb

```diff
@@ -1,8 +1,64 @@
 class AddressesController < ApplicationController
   include CartsHelper
+  protect_from_forgery except: :new
 
   def index
-    @address = current_user.addresses.first
+    @modal_id
+    @addresses = current_user.addresses
     @order_summery = set_order_summery
   end
+
+  def new
+    @address = Address.new
+    @header = "Address"
+  end
+  
+  def create
+    @address = current_user.addresses.new(address_params)
+    if @address.save
+      flash[:notice] = "New address added"
+      redirect_to addresses_path
+    else
+      flash[:alert] = "Error ocuur during create"
+      redirect_to addresses_path
+    end
+  end
+
+  def edit
+    @address = Address.find(params[:id])
+    @header = "Edit Address"
+  end 
+
+  def update
+    @address = Address.find(params[:id])
```

Comment:
If want you use this  "@address = Address.find(params[:id])"  in multiple place, So you should make a method for this and. Avoid repitation